default/AI/ShipDesignAI.py

```diff
@@ -0,0 +1,1602 @@
+"""
+This module deals with the autonomous shipdesign from the AI. The design process is class-based.
+The basic functionality is defined in the class AIShipDesign. The more specialised classes mostly 
+implement the rating function and some additional information for the optimizing algorithms to improve performance. 
+
+Example usage of this module:
+import ShipDesignAI
+myDesign = ShipDesignAI.AIShipDesign_Military()
+myDesign.additional_specifications.enemy_mine_dmg = 10
+best_military_designs = myDesign.optimize_design()  # best designs per planet: (rating,planetID,design_id,cost) tuples
+
+
+Available ship classes:
+- AIShipDesign_Military: basic military ship
```

Comment:
You mean AIShipDesign as classname?

The intention was to make it clearly distinct from "real" ShipDesigns (i.e. universe ingame objects). I'm open for alternative name suggestions but I must disagree with simply removing it.